tests/validators/test_url.py

```diff
@@ -198,29 +252,8 @@ def url_validator_fixture():
     ],
 )
 def test_url_cases(url_validator, url, expected):
-    if isinstance(expected, Err):
-        with pytest.raises(ValidationError) as exc_info:
-            url_validator.validate_python(url)
-        assert exc_info.value.error_count() == 1
-        error = exc_info.value.errors()[0]
-        assert error['type'] == 'url_parsing'
-        assert error['ctx']['error'] == expected.message
-    else:
-        output_url = url_validator.validate_python(url)
-        assert isinstance(output_url, (Url, MultiHostUrl))
-        if isinstance(expected, str):
-            assert str(output_url) == expected
-        else:
-            assert isinstance(expected, dict)
-            output_parts = {}
-            for key in expected:
-                if key == 'str()':
-                    output_parts[key] = str(output_url)
-                elif key.endswith('()'):
-                    output_parts[key] = getattr(output_url, key[:-2])()
-                else:
-                    output_parts[key] = getattr(output_url, key)
-            assert output_parts == expected
+    url_test_case_helper(url, expected, UrlValidatorMode.SCHEMA_VALIDATOR, url_validator)
```

Comment:
better to split into two tests or parametrize with a new param, e.g.

```py
@pytest.mark.parametrize('mode', ['schema_validator', 'url_class'])
@pytest.mark.parametrize('url,expected', ..)
def test_url_cases(url_validator, url, expected, mode):
```

Then you can call `test_url_cases` with mode as an argument below.

Reply:
Ah that's nice - I used another parametrization instead